lib/ark-sdk/gen_sdk/overlay/python/ark_sdk/auth/validator.py

```diff
@@ -0,0 +1,116 @@
+"""Token validation for ARK SDK."""
+
+import asyncio
+import logging
+from typing import Optional, Dict, Any
+import httpx
+from jwt import decode, PyJWKClient
+from jwt.exceptions import InvalidTokenError, ExpiredSignatureError, DecodeError
+
+from .exceptions import TokenValidationError, InvalidTokenError as AuthInvalidTokenError, ExpiredTokenError
+from .config import AuthConfig
+
+logger = logging.getLogger(__name__)
+
+
+class TokenValidator:
+    """Validates JWT tokens using JWKS."""
+    
+    def __init__(self, config: Optional[AuthConfig] = None):
+        self.config = config or AuthConfig()
+        self._jwks_client: Optional[PyJWKClient] = None
+        self._jwks_cache: Dict[str, Any] = {}
+        self._cache_expiry: Optional[float] = None
+    
+    def _get_jwks_client(self) -> PyJWKClient:
+        """Get or create JWKS client."""
+        if self._jwks_client is None and self.config.jwks_url:
+            self._jwks_client = PyJWKClient(
+                self.config.jwks_url,
+                cache_ttl=self.config.jwks_cache_ttl
+            )
+        return self._jwks_client
+    
+    async def validate_token(self, token: str) -> Dict[str, Any]:
+        """
+        Validate a JWT token.
+        
+        Args:
+            token: The JWT token to validate
+            
+        Returns:
+            The decoded token payload
+            
+        Raises:
+            TokenValidationError: If token validation fails
+        """
+        try:
+            # Get the JWKS client
+            jwks_client = self._get_jwks_client()
+            if jwks_client is None:
+                raise TokenValidationError("JWKS URL not configured")
+            
+            # Get the signing key
+            signing_key = jwks_client.get_signing_key_from_jwt(token)
+            
+            # Use OKTA values as primary, JWT values as fallback for backward compatibility
```

Comment:
this looks like breadcrumbs from an earlier change or something? OKTA with JWT as fallback sounds like a mistake